cfdm/read_write/netcdf/netcdfread.py

```diff
@@ -3185,7 +3366,7 @@ def _create_field_or_domain(self, field_ncvar, domain=False):
         ndim = g["variables"][field_ncvar].ndim
 
         if field:
-            if has_dimensions_attr and ndim < 1:
+            if g["CF>=1.9"] and has_dimensions_attr:
```

Comment:
Ah. This is because I belatedly realised that CF-netCDF domain variables do not need to be scalar (["The variable should be a scalar"](https://cfconventions.org/cf-conventions/cf-conventions.html#domain-variables)).

In fact, the following `or ndim >= 1` should be removed, too, I think, for the same reason.

Reply:
See de8ab83a58fa3e0f34ab4cd6f238935dcb668eb9

Reply:
Thanks for explaining. That makes sense. The new commits look fine, so marking as resolved.